src/main/java/com/flightstats/hub/dao/aws/S3AccessMonitor.java

```diff
@@ -89,6 +89,7 @@ private String key() {
             return CompletableFuture.supplyAsync(() -> {
                 try (S3Object s3Object = hubS3Client
                         .getObject(new GetObjectRequest(bucketName, key()))) {
+                    s3Object.getObjectContent().read(new byte[s3Object.getObjectContent().available()]);
```

Comment:
would calling `s3Object.getObjectContent().close()` do roughly the same thing?

Reply:
that would throw, if we explicitly close the connection without reading all the bytes out it blows up.

Reply:
This makes me leery given the unknown size of the bucket. Is the goal here to read the entire stream and discard it, much like an http request? In this case, available() is what's currently in the buffer, not the actual length of the document.

I'd do something like have a 4k buffer and read into it repeatedly until EOF.

Reply:
(If you have IOUtils available, you can readFully into a NullOutputStream to save on code, I believe)